pkg/apis/manager/health/python/health_pb2_grpc.py

```diff
@@ -1,6 +1,5 @@
 # Generated by the gRPC Python protocol compiler plugin. DO NOT EDIT!
 import grpc
-
```

Comment:
Should we exclude files that auto-generated ?

Reply:
I think it's okay to modify them, but am happy to update the pre-commit file if excluding them is preferable. There are quite a few files that fall into this category (assuming there aren't more that don't include the `DO NOT EDIT` warning:

```
❯ grep -r "DO NOT EDIT" . | wc -l
     110
```

Reply:
It looks like pre-commit doesn't let you exclude files based on content, just a regex for the filename, but the generated files have names that truly run the gamut. Given that these specific modifications are superficial and will be overwritten when the files are regenerated anyway, do you mind if we keep them?

Reply:
It might be problematic, since we run `make generate` command to make sure user runs this script before making any API changes.
Can't we use `exclude` parameter ?
https://pre-commit.com/#top_level-exclude

Reply:
We are doing it here: https://github.com/kubeflow/katib/blob/master/.github/workflows/test-go.yaml#L34

Reply:
The problem is that `exclude` accepts a regex and there is no consistent pattern to the filenames of the generate files:

```
> grep -r "DO NOT EDIT" .

./pkg/apis/config/v1beta1/zz_generated.deepcopy.go:// Code generated by controller-gen. DO NOT EDIT.
./pkg/apis/controller/trials/v1beta1/zz_generated.deepcopy.go:// Code generated by deepcopy-gen. DO NOT EDIT.
./pkg/apis/controller/experiments/v1beta1/zz_generated.deepcopy.go:// Code generated by deepcopy-gen. DO NOT EDIT.
./pkg/apis/controller/common/v1beta1/zz_generated.deepcopy.go:// Code generated by deepcopy-gen. DO NOT EDIT.
./pkg/apis/controller/suggestions/v1beta1/zz_generated.deepcopy.go:// Code generated by deepcopy-gen. DO NOT EDIT.
./pkg/apis/manager/health/python/health_pb2.py:# Generated by the protocol buffer compiler.  DO NOT EDIT!
./pkg/apis/manager/health/python/health_pb2_grpc.py:# Generated by the gRPC Python protocol compiler plugin. DO NOT EDIT!
./pkg/apis/manager/health/health_grpc.pb.go:// Code generated by protoc-gen-go-grpc. DO NOT EDIT.
./pkg/apis/manager/health/health.pb.go:// Code generated by protoc-gen-go. DO NOT EDIT.
./pkg/apis/manager/v1beta1/python/api_pb2.py:# Generated by the protocol buffer compiler.  DO NOT EDIT!
./pkg/apis/manager/v1beta1/python/api_pb2_grpc.py:# Generated by the gRPC Python protocol compiler plugin. DO NOT EDIT!
./pkg/apis/manager/v1beta1/api.pb.go:// Code generated by protoc-gen-go. DO NOT EDIT.
./pkg/apis/manager/v1beta1/api_grpc.pb.go:// Code generated by protoc-gen-go-grpc. DO NOT EDIT.
./pkg/apis/v1beta1/openapi_generated.go:// Code generated by main. DO NOT EDIT.
./pkg/mock/v1beta1/util/katibclient/katibclient.go:// Code generated by MockGen. DO NOT EDIT.
./pkg/mock/v1beta1/suggestion/suggestionclient/suggestionclient.go:// Code generated by MockGen. DO NOT EDIT.
./pkg/mock/v1beta1/trial/managerclient/katibmanager.go:// Code generated by MockGen. DO NOT EDIT.
./pkg/mock/v1beta1/db/db.go:// Code generated by MockGen. DO NOT EDIT.
./pkg/mock/v1beta1/api/suggestion.go:// Code generated by MockGen. DO NOT EDIT.
./pkg/mock/v1beta1/api/earlystopping.go:// Code generated by MockGen. DO NOT EDIT.
./pkg/mock/v1beta1/experiment/suggestion/suggestion.go:// Code generated by MockGen. DO NOT EDIT.
./pkg/mock/v1beta1/experiment/manifest/generator.go:// Code generated by MockGen. DO NOT EDIT.
./pkg/client/controller/informers/externalversions/trials/interface.go:// Code generated by informer-gen. DO NOT EDIT.
./pkg/client/controller/informers/externalversions/trials/v1beta1/trial.go:// Code generated by informer-gen. DO NOT EDIT.
./pkg/client/controller/informers/externalversions/trials/v1beta1/interface.go:// Code generated by informer-gen. DO NOT EDIT.
./pkg/client/controller/informers/externalversions/factory.go:// Code generated by informer-gen. DO NOT EDIT.
./pkg/client/controller/informers/externalversions/experiments/interface.go:// Code generated by informer-gen. DO NOT EDIT.
./pkg/client/controller/informers/externalversions/experiments/v1beta1/interface.go:// Code generated by informer-gen. DO NOT EDIT.
./pkg/client/controller/informers/externalversions/experiments/v1beta1/experiment.go:// Code generated by informer-gen. DO NOT EDIT.
./pkg/client/controller/informers/externalversions/internalinterfaces/factory_interfaces.go:// Code generated by informer-gen. DO NOT EDIT.
./pkg/client/controller/informers/externalversions/generic.go:// Code generated by informer-gen. DO NOT EDIT.
./pkg/client/controller/informers/externalversions/suggestions/interface.go:// Code generated by informer-gen. DO NOT EDIT.
./pkg/client/controller/informers/externalversions/suggestions/v1beta1/suggestion.go:// Code generated by informer-gen. DO NOT EDIT.
./pkg/client/controller/informers/externalversions/suggestions/v1beta1/interface.go:// Code generated by informer-gen. DO NOT EDIT.
./pkg/client/controller/applyconfiguration/trials/v1beta1/trial.go:// Code generated by applyconfiguration-gen. DO NOT EDIT.
./pkg/client/controller/applyconfiguration/trials/v1beta1/trialstatus.go:// Code generated by applyconfiguration-gen. DO NOT EDIT.
./pkg/client/controller/applyconfiguration/trials/v1beta1/trialspec.go:// Code generated by applyconfiguration-gen. DO NOT EDIT.
./pkg/client/controller/applyconfiguration/trials/v1beta1/trialcondition.go:// Code generated by applyconfiguration-gen. DO NOT EDIT.
./pkg/client/controller/applyconfiguration/experiments/v1beta1/nasconfig.go:// Code generated by applyconfiguration-gen. DO NOT EDIT.
./pkg/client/controller/applyconfiguration/experiments/v1beta1/optimaltrial.go:// Code generated by applyconfiguration-gen. DO NOT EDIT.
./pkg/client/controller/applyconfiguration/experiments/v1beta1/experimentstatus.go:// Code generated by applyconfiguration-gen. DO NOT EDIT.
./pkg/client/controller/applyconfiguration/experiments/v1beta1/parameterspec.go:// Code generated by applyconfiguration-gen. DO NOT EDIT.
./pkg/client/controller/applyconfiguration/experiments/v1beta1/operation.go:// Code generated by applyconfiguration-gen. DO NOT EDIT.
./pkg/client/controller/applyconfiguration/experiments/v1beta1/trialparameterspec.go:// Code generated by applyconfiguration-gen. DO NOT EDIT.
./pkg/client/controller/applyconfiguration/experiments/v1beta1/trialsource.go:// Code generated by applyconfiguration-gen. DO NOT EDIT.
./pkg/client/controller/applyconfiguration/experiments/v1beta1/feasiblespace.go:// Code generated by applyconfiguration-gen. DO NOT EDIT.
./pkg/client/controller/applyconfiguration/experiments/v1beta1/configmapsource.go:// Code generated by applyconfiguration-gen. DO NOT EDIT.
./pkg/client/controller/applyconfiguration/experiments/v1beta1/experimentspec.go:// Code generated by applyconfiguration-gen. DO NOT EDIT.
./pkg/client/controller/applyconfiguration/experiments/v1beta1/experiment.go:// Code generated by applyconfiguration-gen. DO NOT EDIT.
./pkg/client/controller/applyconfiguration/experiments/v1beta1/graphconfig.go:// Code generated by applyconfiguration-gen. DO NOT EDIT.
./pkg/client/controller/applyconfiguration/experiments/v1beta1/experimentcondition.go:// Code generated by applyconfiguration-gen. DO NOT EDIT.
./pkg/client/controller/applyconfiguration/experiments/v1beta1/trialtemplate.go:// Code generated by applyconfiguration-gen. DO NOT EDIT.
./pkg/client/controller/applyconfiguration/internal/internal.go:// Code generated by applyconfiguration-gen. DO NOT EDIT.
./pkg/client/controller/applyconfiguration/common/v1beta1/algorithmsetting.go:// Code generated by applyconfiguration-gen. DO NOT EDIT.
./pkg/client/controller/applyconfiguration/common/v1beta1/earlystoppingrule.go:// Code generated by applyconfiguration-gen. DO NOT EDIT.
./pkg/client/controller/applyconfiguration/common/v1beta1/filesystempath.go:// Code generated by applyconfiguration-gen. DO NOT EDIT.
./pkg/client/controller/applyconfiguration/common/v1beta1/observation.go:// Code generated by applyconfiguration-gen. DO NOT EDIT.
./pkg/client/controller/applyconfiguration/common/v1beta1/sourcespec.go:// Code generated by applyconfiguration-gen. DO NOT EDIT.
./pkg/client/controller/applyconfiguration/common/v1beta1/collectorspec.go:// Code generated by applyconfiguration-gen. DO NOT EDIT.
./pkg/client/controller/applyconfiguration/common/v1beta1/algorithmspec.go:// Code generated by applyconfiguration-gen. DO NOT EDIT.
./pkg/client/controller/applyconfiguration/common/v1beta1/metricscollectorspec.go:// Code generated by applyconfiguration-gen. DO NOT EDIT.
./pkg/client/controller/applyconfiguration/common/v1beta1/earlystoppingsetting.go:// Code generated by applyconfiguration-gen. DO NOT EDIT.
./pkg/client/controller/applyconfiguration/common/v1beta1/metric.go:// Code generated by applyconfiguration-gen. DO NOT EDIT.
./pkg/client/controller/applyconfiguration/common/v1beta1/filterspec.go:// Code generated by applyconfiguration-gen. DO NOT EDIT.
./pkg/client/controller/applyconfiguration/common/v1beta1/earlystoppingspec.go:// Code generated by applyconfiguration-gen. DO NOT EDIT.
./pkg/client/controller/applyconfiguration/common/v1beta1/metricstrategy.go:// Code generated by applyconfiguration-gen. DO NOT EDIT.
./pkg/client/controller/applyconfiguration/common/v1beta1/objectivespec.go:// Code generated by applyconfiguration-gen. DO NOT EDIT.
./pkg/client/controller/applyconfiguration/common/v1beta1/parameterassignment.go:// Code generated by applyconfiguration-gen. DO NOT EDIT.
./pkg/client/controller/applyconfiguration/utils.go:// Code generated by applyconfiguration-gen. DO NOT EDIT.
./pkg/client/controller/applyconfiguration/suggestions/v1beta1/suggestion.go:// Code generated by applyconfiguration-gen. DO NOT EDIT.
./pkg/client/controller/applyconfiguration/suggestions/v1beta1/suggestionspec.go:// Code generated by applyconfiguration-gen. DO NOT EDIT.
./pkg/client/controller/applyconfiguration/suggestions/v1beta1/suggestionstatus.go:// Code generated by applyconfiguration-gen. DO NOT EDIT.
./pkg/client/controller/applyconfiguration/suggestions/v1beta1/suggestioncondition.go:// Code generated by applyconfiguration-gen. DO NOT EDIT.
./pkg/client/controller/applyconfiguration/suggestions/v1beta1/trialassignment.go:// Code generated by applyconfiguration-gen. DO NOT EDIT.
./pkg/client/controller/listers/trials/v1beta1/trial.go:// Code generated by lister-gen. DO NOT EDIT.
./pkg/client/controller/listers/trials/v1beta1/expansion_generated.go:// Code generated by lister-gen. DO NOT EDIT.
./pkg/client/controller/listers/experiments/v1beta1/experiment.go:// Code generated by lister-gen. DO NOT EDIT.
./pkg/client/controller/listers/experiments/v1beta1/expansion_generated.go:// Code generated by lister-gen. DO NOT EDIT.
./pkg/client/controller/listers/suggestions/v1beta1/suggestion.go:// Code generated by lister-gen. DO NOT EDIT.
./pkg/client/controller/listers/suggestions/v1beta1/expansion_generated.go:// Code generated by lister-gen. DO NOT EDIT.
./pkg/client/controller/clientset/versioned/scheme/register.go:// Code generated by client-gen. DO NOT EDIT.
./pkg/client/controller/clientset/versioned/scheme/doc.go:// Code generated by client-gen. DO NOT EDIT.
./pkg/client/controller/clientset/versioned/clientset.go:// Code generated by client-gen. DO NOT EDIT.
./pkg/client/controller/clientset/versioned/typed/trials/v1beta1/trial.go:// Code generated by client-gen. DO NOT EDIT.
./pkg/client/controller/clientset/versioned/typed/trials/v1beta1/generated_expansion.go:// Code generated by client-gen. DO NOT EDIT.
./pkg/client/controller/clientset/versioned/typed/trials/v1beta1/trials_client.go:// Code generated by client-gen. DO NOT EDIT.
./pkg/client/controller/clientset/versioned/typed/trials/v1beta1/doc.go:// Code generated by client-gen. DO NOT EDIT.
./pkg/client/controller/clientset/versioned/typed/trials/v1beta1/fake/fake_trial.go:// Code generated by client-gen. DO NOT EDIT.
./pkg/client/controller/clientset/versioned/typed/trials/v1beta1/fake/doc.go:// Code generated by client-gen. DO NOT EDIT.
./pkg/client/controller/clientset/versioned/typed/trials/v1beta1/fake/fake_trials_client.go:// Code generated by client-gen. DO NOT EDIT.
./pkg/client/controller/clientset/versioned/typed/experiments/v1beta1/experiments_client.go:// Code generated by client-gen. DO NOT EDIT.
./pkg/client/controller/clientset/versioned/typed/experiments/v1beta1/generated_expansion.go:// Code generated by client-gen. DO NOT EDIT.
./pkg/client/controller/clientset/versioned/typed/experiments/v1beta1/doc.go:// Code generated by client-gen. DO NOT EDIT.
./pkg/client/controller/clientset/versioned/typed/experiments/v1beta1/experiment.go:// Code generated by client-gen. DO NOT EDIT.
./pkg/client/controller/clientset/versioned/typed/experiments/v1beta1/fake/doc.go:// Code generated by client-gen. DO NOT EDIT.
./pkg/client/controller/clientset/versioned/typed/experiments/v1beta1/fake/fake_experiments_client.go:// Code generated by client-gen. DO NOT EDIT.
./pkg/client/controller/clientset/versioned/typed/experiments/v1beta1/fake/fake_experiment.go:// Code generated by client-gen. DO NOT EDIT.
./pkg/client/controller/clientset/versioned/typed/common/v1beta1/generated_expansion.go:// Code generated by client-gen. DO NOT EDIT.
./pkg/client/controller/clientset/versioned/typed/common/v1beta1/doc.go:// Code generated by client-gen. DO NOT EDIT.
./pkg/client/controller/clientset/versioned/typed/common/v1beta1/common_client.go:// Code generated by client-gen. DO NOT EDIT.
./pkg/client/controller/clientset/versioned/typed/common/v1beta1/fake/fake_common_client.go:// Code generated by client-gen. DO NOT EDIT.
./pkg/client/controller/clientset/versioned/typed/common/v1beta1/fake/doc.go:// Code generated by client-gen. DO NOT EDIT.
./pkg/client/controller/clientset/versioned/typed/suggestions/v1beta1/suggestion.go:// Code generated by client-gen. DO NOT EDIT.
./pkg/client/controller/clientset/versioned/typed/suggestions/v1beta1/generated_expansion.go:// Code generated by client-gen. DO NOT EDIT.
./pkg/client/controller/clientset/versioned/typed/suggestions/v1beta1/suggestions_client.go:// Code generated by client-gen. DO NOT EDIT.
./pkg/client/controller/clientset/versioned/typed/suggestions/v1beta1/doc.go:// Code generated by client-gen. DO NOT EDIT.
./pkg/client/controller/clientset/versioned/typed/suggestions/v1beta1/fake/fake_suggestion.go:// Code generated by client-gen. DO NOT EDIT.
./pkg/client/controller/clientset/versioned/typed/suggestions/v1beta1/fake/fake_suggestions_client.go:// Code generated by client-gen. DO NOT EDIT.
./pkg/client/controller/clientset/versioned/typed/suggestions/v1beta1/fake/doc.go:// Code generated by client-gen. DO NOT EDIT.
./pkg/client/controller/clientset/versioned/fake/register.go:// Code generated by client-gen. DO NOT EDIT.
./pkg/client/controller/clientset/versioned/fake/clientset_generated.go:// Code generated by client-gen. DO NOT EDIT.
./pkg/client/controller/clientset/versioned/fake/doc.go:// Code generated by client-gen. DO NOT EDIT.
```

It's weird though because the gate is passing even with the pre-commit modifications!
https://github.com/kubeflow/katib/actions/runs/9941076698/job/27459292540?pr=2242

Also, we can probably relocate the golangci-lint execution into pre-commit in a subsequent PR.

Reply:
>It's weird though because the gate is passing even with the pre-commit modifications!

It looks like a bug, and this script just doesn't work: https://github.com/kubeflow/katib/blob/master/hack/verify-generated-codes.sh

Can we use subpath and appropriate regex in exclude ?
https://pre-commit.com/#regular-expressions

Reply:
For example, I can see that KFP does it: https://github.com/kubeflow/pipelines/blob/master/.pre-commit-config.yaml#L48

Reply:
Thanks, @andreyvelich! Any objection to following KFP's example and just listing the full paths? With 113 unique generated files, it's going to take forever to generate the 30 or so regexes that will be required to cover them all and will arguably reduce readability (regex is a write-only language 😂 ).

Reply:
I guess, we don't need to list all 113 unique files, just find common patterns.
E.g.
```
exclude: (/pkg/client/|/pkg/mock/|/pkg/apis/manager|*zz_generated.deepcopy.go|sdk/python/v1beta1/kubeflow/katib/models/)

Reply:
You're absolutely, right! Sorry, I was just being lazy. Fix incoming.

Reply:
Done! Also squashed and removed all the generated files, then validated that it's excluding correctly by running `pre-commit run --all-files` and confirming that none of the generated files get modified.